Integrations/python/deephaven/Types.py

```diff
@@ -0,0 +1,311 @@
+#
+# Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
+#
+"""
+Basic Deephaven table related data types
+"""
+
+import jpy
+import wrapt
+import sys
+import collections
+from typing import NewType
+
+# None until the first _defineSymbols() call
+_table_tools_ = None
+_col_def_ = None
+_python_tools_ = None
+_qst_col_header_ = None
+_qst_column_ = None
+_qst_newtable_ = None
+_qst_type_ = None
+_table_ = None
+
+def _defineSymbols():
+    if not jpy.has_jvm():
+        raise SystemError("No java functionality can be used until the JVM has been initialized through the jpy module")
+
+    global _table_tools_, _col_def_, _python_tools_, _qst_col_header_, _qst_column_, \
+        _qst_newtable_, _qst_type_, _table_
+    if _table_tools_ is None:
+        # This will raise an exception if the desired object is not the classpath
+        _table_tools_ = jpy.get_type("io.deephaven.db.tables.utils.TableTools")
+        _col_def_ = jpy.get_type("io.deephaven.db.tables.ColumnDefinition")
+        _python_tools_ = jpy.get_type("io.deephaven.integrations.python.PythonTools")
+        _qst_col_header_ = jpy.get_type("io.deephaven.qst.column.header.ColumnHeader")
+        _qst_column_ =  jpy.get_type("io.deephaven.qst.column.Column")
+        _qst_newtable_ = jpy.get_type("io.deephaven.qst.table.NewTable")
+        _qst_type_ = jpy.get_type("io.deephaven.qst.type.Type")
+        _table_ = jpy.get_type("io.deephaven.db.tables.Table")
+
+# every method that depends on symbols defined via _defineSymbols() should be decorated with @_passThrough
+@wrapt.decorator
+def _passThrough(wrapped, instance, args, kwargs):
+    """
+    For decoration of module methods, to define necessary symbols at runtime
+
+    :param wrapped: the method to be decorated
+    :param instance: the object to which the wrapped function was bound when it was called
+    :param args: the argument list for `wrapped`
+    :param kwargs: the keyword argument dictionary for `wrapped`
+    :return: the decorated version of the method
+    """
+
+    _defineSymbols()
+    return wrapped(*args, **kwargs)
+
+try:
+    _defineSymbols()
+except Exception as e:
+    pass
+
+
+# Represents a Deephaven column data type.
+DataType = NewType('DataType', _qst_type_)
+
+# For more involved types, you can always use the string representation
+# of the Java class (Class.getName()) to get a python type for it.
+@_passThrough
+def typeFromJavaClassName(name : str):
+    """
+    Get the column data type for the corresponding Java type string reprensentation
+    The string provided should match the output in Java for Class.getName()
+    for a class visible to the main ClassLoader in the Deephaven engine in use.
+    """
```

Comment:
These comments are not correct. `String[].class.getName` is *not* "java.lang.String[]". `name` here is a Deephaven-specific "pretty" version.

Reply:
Fixed.